libraries/exoplayer_dash/src/main/java/androidx/media3/exoplayer/dash/manifest/DashManifestParser.java

```diff
@@ -819,6 +854,10 @@ protected Format buildFormat(
         codecs = MimeTypes.CODEC_E_AC3_JOC;
       }
     }
+    if (isDolbyVisionFormat(codecs, supplementalCodecs)) {
+      sampleMimeType = MimeTypes.VIDEO_DOLBY_VISION;
+      codecs = supplementalCodecs != null ? supplementalCodecs : codecs;
```

Comment:
For Dolby Vision profile 5, which is not backward compatible so that there is no supplemental codecs field in manifest. In this case, `supplementalCodecs` is null. I added this logic when I found issue in my testing. :)

See https://professionalsupport.dolby.com/s/article/How-to-signal-Dolby-Vision-in-MPEG-DASH?language=en_US, section 3.1 for details.

Reply:
Sorry, I missed the earlier return for dvhe and dvh1. Thanks for the clarification!